eclipsePlugin/src/de/tobject/findbugs/builder/PDEClassPathGenerator.java

```diff
@@ -90,18 +90,16 @@ private static Set<String> createJavaClasspath(IJavaProject javaProject) {
             for (IClasspathEntry entry : rawClasspath) {
                 if (entry.getEntryKind() == IClasspathEntry.CPE_CONTAINER) {
                     IClasspathContainer classpathContainer = JavaCore.getClasspathContainer(entry.getPath(), javaProject);
-                    if (classpathContainer != null) {
-                        if (classpathContainer instanceof JREContainer) {
-                            IClasspathEntry[] classpathEntries = classpathContainer.getClasspathEntries();
-                            for (IClasspathEntry iClasspathEntry : classpathEntries) {
-                                IPath path = iClasspathEntry.getPath();
-                                // smallest possible fix for #1228 Eclipse plugin always uses host VM to resolve JDK classes
-                                if (isValidPath(path) &&
-                                        ("rt.jar".equals(path.lastSegment())
-                                                || "jrt-fs.jar".equals(path.lastSegment())
-                                                || "jce.jar".equals(path.lastSegment()))) {
-                                    classPath.add(path.toOSString());
-                                }
+                    if (classpathContainer instanceof JREContainer) {
+                        IClasspathEntry[] classpathEntries = classpathContainer.getClasspathEntries();
+                        for (IClasspathEntry iClasspathEntry : classpathEntries) {
+                            IPath path = iClasspathEntry.getPath();
+                            // smallest possible fix for #1228 Eclipse plugin always uses host VM to resolve JDK classes
+                            if (isValidPath(path) &&
+                                    ("rt.jar".equals(path.lastSegment())
+                                            || "jrt-fs.jar".equals(path.lastSegment())
+                                            || "jce.jar".equals(path.lastSegment()))) {
+                                classPath.add(path.toOSString());
```

Comment:
This is the only modification, where the fix is not the exact answer to the Sonar Analyses. It comes from here: [Merge this if statement with the enclosing one.](https://sonarcloud.io/project/issues?createdAfter=2023-01-01&resolved=false&id=com.github.spotbugs.spotbugs&open=AYuNJIc12YKIFvInB8dl), but the instanceof check makes the null check redundant.